packages/rum-core/src/common/compress.js

```diff
@@ -0,0 +1,294 @@
+        break
+      case 'response':
+        compressed.r = compressResponse(value)
+        break
+    }
+  }
+  return compressed
+}
+
+function compressContext(context) {
+  if (!context) {
+    return null
+  }
+  const compressed = {}
+  for (const key of Object.keys(context)) {
+    const value = context[key]
+    switch (key) {
+      case 'page':
+        compressed.p = {
+          rf: value.referer,
+          url: value.url
+        }
+        break
+      case 'http':
+        compressed.h = compressHTTP(value)
+        break
+      case 'response':
+        compressed.r = compressResponse(value)
+        break
+      case 'user':
+        compressed.u = {
+          id: value.id,
+          un: value.username,
+          em: value.email
+        }
+        break
+      case 'destination':
+        const { service } = value
+        compressed.dt = {
+          se: {
+            n: service.name,
+            t: service.type,
+            rc: service.resource
+          },
+          ad: value.address,
+          po: value.port
+        }
+        break
+      case 'custom':
+        compressed.cu = value
+        break
+    }
+  }
+  return compressed
+}
+
+function compressMarks(marks) {
+  if (!marks) {
+    return null
+  }
+  const { navigationTiming, agent } = marks
+  const compressed = { nt: {}, a: {} }
+
+  COMPRESSED_NAV_TIMING_MARKS.forEach((mark, index) => {
+    const mapping = NAVIGATION_TIMING_MARKS[index]
+    compressed.nt[mark] = navigationTiming[mapping]
+  })
+
+  compressed.a = {
+    fb: compressed.nt.rs,
+    di: compressed.nt.di,
+    dc: compressed.nt.dc
+  }
+  const fp = agent.firstContentfulPaint
+  const lp = agent.largestContentfulPaint
+  if (fp) {
+    compressed.a.fp = fp
+  }
+  if (lp) {
+    compressed.a.lp = lp
+  }
+
+  return compressed
+}
+
+export function compressMetadata(metadata) {
+  const { service, labels } = metadata
+  const { agent, language } = service
+  return {
+    se: {
+      n: service.name,
+      ve: service.version,
+      a: {
+        n: agent.name,
+        ve: agent.version
+      },
+      la: {
+        n: language.name
+      },
+      en: service.environment
+    },
+    l: labels
+  }
+}
+
+export function compressTransaction(transaction) {
+  const spans = transaction.spans.map(span => {
+    const spanData = {
+      id: span.id,
+      n: span.name,
+      t: span.type,
+      s: span.start,
+      d: span.duration,
+      c: compressContext(span.context)
+    }
+    /**
+     * Set parentId only for spans that are child of other spans
+     * and not transaction as it will be inferred
+     */
+    if (span.parent_id !== transaction.id) {
+      spanData.pid = span.parent_id
+    }
+    /**
+     * Set the below fields only when they are truthy or exists
+     */
+    if (span.sync === true) {
+      spanData.sy = true
+    }
+    if (span.subtype) {
+      spanData.su = span.subtype
+    }
+    if (span.action) {
+      spanData.ac = span.action
```

Comment:
Did we used to set empty string on these fields?

Reply:
No its undefined by default.